quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusCodeActions.java

```diff
@@ -0,0 +1,98 @@
+import com.redhat.quarkus.commons.QuarkusProjectInfo;
+import com.redhat.quarkus.ls.commons.BadLocationException;
+import com.redhat.quarkus.ls.commons.CodeActionFactory;
+import com.redhat.quarkus.model.PropertiesModel;
+import com.redhat.quarkus.model.PropertyKey;
+import com.redhat.quarkus.settings.QuarkusFormattingSettings;
+import com.redhat.quarkus.utils.PositionUtils;
+
+/**
+ * The Quarkus code actions
+ * 
+ * @author Angelo ZERR
+ *
+ */
+class QuarkusCodeActions {
+
+	private static final float MAX_DISTANCE_DIFF_RATIO = 0.1f;
+
+	private static final Logger LOGGER = Logger.getLogger(QuarkusCodeActions.class.getName());
+
+	/**
+	 * Returns code actions for the given diagnostics of the application.properties
+	 * <code>document</code> by using the given Quarkus properties metadata
+	 * <code>projectInfo</code>.
+	 * 
+	 * @param context            the code action context
+	 * @param range              the range
+	 * @param document           the properties model.
+	 * @param projectInfo        the Quarkus properties
+	 * @param formattingSettings the formatting settings.
+	 * @return the result of the code actions.
+	 */
+	public List<CodeAction> doCodeActions(CodeActionContext context, Range range, PropertiesModel document,
+			QuarkusProjectInfo projectInfo, QuarkusFormattingSettings formattingSettings) {
+		List<CodeAction> codeActions = new ArrayList<>();
+		if (context.getDiagnostics() != null) {
+			// Loop for all diagnostics
+			for (Diagnostic diagnostic : context.getDiagnostics()) {
+				if (ValidationType.unknown.name().equals(diagnostic.getCode())) {
+					// Manage code action for unknown
+					doCodeActionsForUnknown(diagnostic, document, projectInfo, codeActions);
+				}
+			}
+		}
+		return codeActions;
+	}
+
+	private void doCodeActionsForUnknown(Diagnostic diagnostic, PropertiesModel document,
+			QuarkusProjectInfo projectInfo, List<CodeAction> codeActions) {
+		try {
+			// Get property name by using the diagnostic range
+			PropertyKey propertyKey = (PropertyKey) document.findNodeAt(diagnostic.getRange().getStart());
+			String propertyName = propertyKey.getPropertyName();
+			// Loop for each metata property
+			for (ExtendedConfigDescriptionBuildItem metaProperty : projectInfo.getProperties()) {
+				// Check if the property name is similar to the metadata name
+				if (isSimilar(propertyName, metaProperty.getPropertyName())) {
+					Range range = PositionUtils.createRange(propertyKey);
+					CodeAction replaceAction = CodeActionFactory.replace(
+							"Did you mean '" + metaProperty.getPropertyName() + "' ?", range,
+							metaProperty.getPropertyName(), document.getDocument(), diagnostic);
+					codeActions.add(replaceAction);
+				}
+			}
+		} catch (BadLocationException e) {
+			LOGGER.log(Level.SEVERE, "In QuarkusCodeActions, position error", e);
+		}
+	}
+
+	private static boolean isSimilar(String left, String right) {
+		int threshold = Math.round(MAX_DISTANCE_DIFF_RATIO * left.length());

```

Comment:
does it make sense to determine the threshold as 10% of the length of a word?

Alos it should be the distance from the target value, which is `right` here.

Reply:
It should be fixed.